packages/blade/src/components/Accordion/__tests__/Accordion.web.test.tsx

```diff
@@ -0,0 +1,196 @@
+import assertAccessible from '~src/_helpers/testing/assertAccessible.web';
+import { Button } from '~components/Button';
+
+describe('<Accordion />', () => {
+  it('should render', () => {
+    const { container } = renderWithTheme(
+      <Accordion>
+        <AccordionItem
+          title="How can I setup Route?"
+          description="You can use Razorpay Route from the Dashboard or using APIs to transfer money to customers. You may also check our docs for detailed instructions."
+        />
+        <AccordionItem
+          title="How can I setup QR Codes?"
+          description="Just use Razorpay. You may also check our docs for detailed instructions. Please use the search functionality to ask your queries."
+        />
+        <AccordionItem title="How can I setup Subscriptions?" description="Just use Razorpay." />
+      </Accordion>,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should expand and close in uncontrolled behavior', async () => {
+    const onExpandChange = jest.fn();
+    const button1 = 'How can I setup Route?';
+    const description1 =
+      'You can use Razorpay Route from the Dashboard or using APIs to transfer money to customers. You may also check our docs for detailed instructions.';
+    const button2 = 'How can I setup QR Codes?';
+    const description2 =
+      'Just use Razorpay. You may also check our docs for detailed instructions. Please use the search functionality to ask your queries.';
+
+    const { getByRole, getByText, getAllByRole } = renderWithTheme(
+      <Accordion onExpandChange={onExpandChange}>
+        <AccordionItem title={button1} description={description1} />
+        <AccordionItem title={button2} description={description2} />
+        <AccordionItem title="How can I setup Subscriptions?" description="Just use Razorpay." />
+      </Accordion>,
+    );
+
+    const user = userEvent.setup();
+
+    // Clicking first accordion item
+    const trigger1 = getByRole('button', { name: button1 });
+    expect(trigger1).toHaveAttribute('aria-expanded', 'false');
+    expect(getByText(description1)).not.toBeVisible();
+    expect(getAllByRole('region', { hidden: true })[0]).toHaveAttribute('aria-hidden', 'true');
+
+    await user.click(trigger1);
+    // wait for animation to finish
+    await waitFor(() => expect(getByText(description1)).toBeVisible());
+
+    expect(trigger1).toHaveAttribute('aria-expanded', 'true');
+    expect(getAllByRole('region')[0]).toHaveAttribute('aria-hidden', 'false');
+
+    expect(onExpandChange).toHaveBeenCalledTimes(1);
+    expect(onExpandChange).toHaveBeenLastCalledWith({ expandedIndex: 0 });
+
+    // Clicking second accordion item
+    const trigger2 = getByRole('button', { name: button2 });
+    expect(trigger2).toHaveAttribute('aria-expanded', 'false');
+    expect(getByText(description2)).not.toBeVisible();
+    expect(getAllByRole('region', { hidden: true })[1]).toHaveAttribute('aria-hidden', 'true');
```

Comment:
getByRole(region) is referring to the accordion description or accordion item?

Reply:
If it is AccordionItem shouldn't this be button? 🤔

Reply:
It actually refers to the expanding / collapsing region from user perspective. In code this refers to `CollapsibleBody`